version: '3.8'

services:
  postgres:
    image: postgres:15.5
    container_name: postgres-test
    environment:
      POSTGRES_DB: blogApp
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: 1234
    ports:
      - "6432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - blogapp-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d blogApp"]
      interval: 30s
      timeout: 10s
      retries: 3

  pgadmin:
    image: dpage/pgadmin4:latest
    container_name: blogapp-pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@blogapp.com
      PGADMIN_DEFAULT_PASSWORD: admin123
    ports:
      - "8081:80"
    volumes:
      - pgadmin_data:/var/lib/pgadmin
    networks:
      - blogapp-network
    depends_on:
      postgres:
        condition: service_healthy

volumes:
  postgres_data:
  pgadmin_data:

networks:
  blogapp-network:
    driver: bridge 